implementations/rust/ockam/ockam_api/src/authenticator/direct.rs

```diff
@@ -0,0 +1,419 @@
+pub mod types;
+
+use crate::auth::types::Attributes;
+use crate::signer::types::{Credential, IdentityId};
+use crate::util::response;
+use crate::{assert_request_match, assert_response_match, Cbor};
+use crate::{signer, Timestamp};
+use crate::{Error, Method, Request, RequestBuilder, Response, ResponseBuilder, Status};
+use core::time::Duration;
+use core::{fmt, str};
+use minicbor::encode::write::Cursor;
+use minicbor::{Decoder, Encode};
+use ockam_core::errcode::{Kind, Origin};
+use ockam_core::{self, Address, Result, Route, Routed, Worker};
+use ockam_identity::authenticated_storage::AuthenticatedStorage;
+use ockam_identity::{IdentityIdentifier, IdentitySecureChannelLocalInfo};
+use ockam_node::Context;
+use tracing::{trace, warn};
+use types::{AddEnroller, AddMember, Placeholder};
+
+// storage scopes:
+const ENROLLER: &str = "enroller";
+const MEMBER: &str = "member";
+
+const MAX_VALIDITY: Duration = Duration::from_secs(2 * 3600);
+
+#[derive(Debug)]
+pub struct Server<S, T> {
+    m_store: S, // member store
+    e_store: T, // enroller store
+    signer: signer::Client,
```

Comment:
Is corresponding Server meant to be on the same node? If yes, can we use it directly without this excess Client hop?

Reply:
It is meant to be on the same node. I have some local changes which I can turn into a PR that allows signature validation with just the reference to `Identity` and the `AuthenticatedStorage`.

Reply:
That would be great. Maybe it could be wrapped into a separate struct, which would be used from both here and `Server`

Reply:
There is this: https://github.com/build-trust/ockam/commit/0d8e9b287ddd6e661cfc199f1acb6dfb05b0ab0e. If we make it a struct, would it not be identical to the existing `signer::Server`?